File: 0009-palindrome-number/0009-palindrome-number.py
class Solution:
    def isPalindrome(self, x: int) -> bool:
        if x < 0:
            return False

        temp = x
        reversed = 0

        while temp != 0:
            reversed = reversed*10 + temp%10
            temp = temp // 10
        
        if reversed == x:
            return True
        return False
        

File: 0009-palindrome-number/0009-palindrome-number.cpp
class Solution {
public:
    bool isPalindrome(int x) {
        int org = x;
        unsigned int reversed = 0;

        if (x<0)
            return false;

        while (x!=0){
            reversed = reversed*10 + (int)(x%10);
            x = (int)(x/10);
        }

        if (org != reversed) {
            return false;
        }
        return true;
    }
};